### ISSUE TITLE
Few questions about training

### ISSUE THREAD

**Issue by saharshleo** [ttengwang/PDVC#30]
Hello @ttengwang ,
I am trying to train your model from scratch (just for learning purpose). However I am facing few issues:
1. the `train_caption_file` or `val_caption_file` does not have `labels`, which is being used in video_dataset.py (also in class loss). Am I using some wrong file?
2. I tried with labels from [action_proposal dataset](http://activity-net.org/challenges/2021/tasks/anet_localization.html) (with captioning related part removed), but the `loss_ce` doesn't decrease at all, both in train and val (did you face any issues like this?). Also the loss_ce is coming in ranges of 300-400.
3. How many epochs you trained before getting decent captions?

**Comment by ttengwang** [comment]
1. Since we consider localization as a binary classification, so all `action_labels` is set to 0 (0 for foreground, 1 for background). If you want to perform multi-class classification, add annotations of `action_labels` in `train_caption_file` and `val_caption_file`.
2. I tried to train ActivityNet Captions with the captioning head removed, and the loss and performance are reasonable. The configs are at https://github.com/ttengwang/PDVC/blob/main/cfgs/anet_c3d_props.yml
3. 5 epochs for readable captions, and 10-25 epochs for best performance (It depends on different video features)

**Comment by saharshleo** [comment]
Thank you for your reply,
Can you please provide the file with action labels in it

**Comment by ttengwang** [comment]
The Activitynet Captions datasets were labeled by different annotators with [ActivityNet 1.3](http://activity-net.org/challenges/2021/tasks/anet_localization.html). So there are no official annotations of action labels for Activitynet Captions. Sorry that I didn't train the model on ActivityNet 1.3, so you may prepare a new `train_caption_file` by yourself.

**Comment by saharshleo** [comment]
Hey, I tried with your given anet_c3d_props.yml config, It is now training better and train losses are decreasing steadily.
However val losses are almost constant, can you suggest what parameters I should check or tweak (given that I am training for 200 classes)

**Comment by ttengwang** [comment]
Also, try to set  `bbox_loss_coef` >0.

**Comment by saharshleo** [comment]
Ok thank you,
Keeping `bbox_loss_coef = 0` means you are not using it?,  

This are the current parameters:
```py
batch_size = 16
lr = 1e-4
lr_drop = 200
weight_decay = 1e-4

num_classes = 200
num_queries = 100
d_model = 512

cls_loss_coef = 1
counter_loss_coef = 2
bbox_loss_coef = 5
giou_loss_coef = 2
self_iou_loss_coef = 2
mask_prediction_coef = 2
eos_coef = 0.1

matcher.cost_class = 1 
matcher.cost_segment = 5 
matcher.cost_giou = 2
matcher.cost_alpha = 0.25
matcher.cost_gamma = 2.0

# Deformable Detr
num_heads = 8
num_feature_levels = 4
dec_n_points = 4
enc_n_points = 4
enc_layers = 6
dec_layers = 6
transformer_ff_dim = 2048
```

Result:
![image](https://user-images.githubusercontent.com/50948120/170353986-34ecf6bd-7714-40ad-b654-e7bf71107105.png)
![image](https://user-images.githubusercontent.com/50948120/170354045-65d54b41-72af-4bca-bfcf-6b2179c40770.png)
